plutus-ledger-api/src/Plutus/V1/Ledger/Api.hs

```diff
@@ -0,0 +1,196 @@
+import           Data.ByteString.Short
+import           Data.Either
+import           Data.Map
+import qualified Data.Text                                          as Text
+import           Data.Text.Prettyprint.Doc
+import           Data.Tuple
+import qualified Flat
+import qualified Language.PlutusCore                                as PLC
+import qualified Language.PlutusCore.Constant                       as PLC
+import qualified Language.PlutusCore.DeBruijn                       as PLC
+import           Language.PlutusCore.Evaluation.Machine.ExBudgeting (CostModel, ExBudget (..))
+import qualified Language.PlutusCore.Evaluation.Machine.ExBudgeting as PLC
+import           Language.PlutusCore.Evaluation.Machine.ExMemory    (ExCPU (..), ExMemory (..))
+import qualified Language.PlutusCore.MkPlc                          as PLC
+import           Language.PlutusCore.Pretty
+import           Language.PlutusTx                                  (Data (..), IsData (..))
+import qualified Language.PlutusTx.Lift                             as PlutusTx
+import qualified Language.UntypedPlutusCore                         as UPLC
+import qualified Language.UntypedPlutusCore.Evaluation.Machine.Cek  as UPLC
+import           Plutus.V1.Ledger.Address
+import           Plutus.V1.Ledger.Bytes
+import           Plutus.V1.Ledger.Contexts
+import           Plutus.V1.Ledger.Crypto
+import           Plutus.V1.Ledger.Interval
+import           Plutus.V1.Ledger.Scripts                           hiding (Script)
+import qualified Plutus.V1.Ledger.Scripts                           as Scripts
+import           Plutus.V1.Ledger.Slot
+
+{- Note [Abstract types in the ledger API]
+We need to support old versions of the ledger API as we update the code that it depends on. You
+might think that we should therefore make the types that we expose abstract, and only expose
+specific functions for constructing and working with them. However the situation is slightly
+different for us.
+
+Normally, when you are in this situation, you want to retain the same *interface* as the old version,
+but with the new types and functions underneath. Abstraction lets you do this easily. But we actually
+want to keep the old *implementation*, because things really have to work the same, bug-for-bug. And
+the types have to translate into Plutus Core in exactly the same way, and so on.
+
+So we're going to end up with multiple versions of the types and functions that we expose here, even
+internally. That means we don't lose anything by exposing all the details: we're never going to remove
+anything, we're just going to create new versions.
+-}
+
+-- | Check if a 'Script' is "valid". At the moment this just means "deserialises correctly", which in particular
+-- implies that it is (almost certainly) an encoded script and cannot be interpreted as some other kind of encoded data.
+validateScript :: Script -> Bool
+validateScript = isRight . Flat.unflat @Scripts.Script . fromShort
```

Comment:
I am worried that the name of this function is very similar to a script validator and thus can be confusing: https://docs.cardano.org/projects/plutus/en/latest/tutorials/basic-validators.html